docs(service-worker): improve service worker guides (#43508)
angular/angular: aio/content/guide/service-worker-intro.md

Angular's service worker is designed to optimize the end user experience of using an application over a slow or unreliable network connection, while also minimizing the risks of serving outdated content.

> The Angular service worker's behavior follows these design goals:

Reviewer: The points below are not design goals :grin:
They describe the behavior (which follows the design goal mentioned above).

Author: ah ok :sweat_smile::+1: 

Still I don't think the wording then is on point here... :sweat: 

Would you mind if modified it to something like:
```
The Angular service worker follows this design goal by considering the following:
```

or something of that sort? :thinking:

Reviewer: Fair enough :smiley:
How about:

```suggestion
To achieve this, the Angular service worker follows these guidelines:
```

Author: yeah! :heart_eyes::+1:

* Caching an application is like installing a native application. The application is cached as one unit, and all files update together.
* A running application continues to run with the same version of all files. It does not suddenly start receiving cached files from a newer version, which are likely incompatible.
* When users refresh the application, they see the latest fully cached version. New tabs load the latest cached code.
* Updates happen in the background, relatively quickly after changes are published. The previous version of the application is served until an update is installed and ready.
* The service worker conserves bandwidth when possible. Resources are only downloaded if they've changed.

> To support these behaviors, the Angular service worker loads a *manifest* file from the server. The file, called the `ngsw.json` (not to be confused with the `manifest.webmanifest` file), describes the resources to cache and includes hashes of every file's contents. When an update to the application is deployed, the contents of the manifest change, informing the service worker that a new version of the application should be downloaded and cached. This manifest is generated from a CLI-generated configuration file called `ngsw-config.json`.

Reviewer: Good call :100:
Maybe call the other file "web app manifest" and link to [MDN](https://developer.mozilla.org/en-US/docs/Web/Manifest):

```suggestion
To support these behaviors, the Angular service worker loads a *manifest* file from the server. The file, called `ngsw.json` (not to be confused with the [web app manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest)), describes the resources to cache and includes hashes of every file's contents. When an update to the application is deployed, the contents of the manifest change, informing the service worker that a new version of the application should be downloaded and cached. This manifest is generated from a CLI-generated configuration file called `ngsw-config.json`.
```
